app/javascript/crayons/MentionAutocompleteTextArea/MentionAutocompleteTextArea.jsx

```diff
@@ -15,30 +15,51 @@ const MIN_SEARCH_CHARACTERS = 2;
 const MAX_RESULTS_DISPLAYED = 6;
 
 /**
- * Helper function to copy all styles and attributes from the original textarea to new autocomplete textarea before removing the original node
+ * Helper function to copy all styles and attributes from the original textarea to new autocomplete textareas before removing the original node
  *
- * @param {element} originalNodeToReplace The DOM element that should be replaced
- * @param {element} newNode The DOM element that will receive all attributes and styles of the original node.
+ * @param {object} options
+ * @param {element} options.originalNodeToReplace The DOM element that should be replaced
+ * @param {element} options.plainTextArea The DOM element to be used in the 'non-autosuggest' state. It will receive all attributes and styles of the original node.
+ * @param {element} options.comboboxTextArea The DOM element to be used in the 'autosuggest' state. It will receive all attributes and styles of the original node.
  */
-const replaceTextArea = (originalNodeToReplace, newNode) => {
-  // Make sure all attributes are copied to the autocomplete textarea
+const replaceTextArea = ({
+  originalNodeToReplace,
+  plainTextArea,
+  comboboxTextArea,
+}) => {
+  const newNodes = [plainTextArea, comboboxTextArea];
+
+  // Make sure all attributes are copied to the autocomplete & plain textareas
   const attributes = originalNodeToReplace.attributes;
   Object.keys(attributes).forEach((attributeKey) => {
-    newNode.setAttribute(
-      attributes[attributeKey].name,
-      attributes[attributeKey].value,
-    );
+    newNodes.forEach((node) => {
+      node.setAttribute(
+        attributes[attributeKey].name,
+        attributes[attributeKey].value,
+      );
+    });
   });
 
-  // Make sure all styles are copied to the autocomplete textarea
-  newNode.style.cssText = document.defaultView.getComputedStyle(
+  // Make sure all styles are copied to the autocomplete & plain textareas
+  const cssText = document.defaultView.getComputedStyle(
     originalNodeToReplace,
     '',
   ).cssText;
 
-  // We need to manually remove the element, as Preact's diffing algorithm won't replace it in render
+  newNodes.forEach((node) => {
```

Comment:
Refactor (blocking): Instead of two `forEach` loops in the `Object.keys.forEach` loop, consolidate them as one `forEach` and set the `cssText` before the loop starts.

Reply:
There's only one `forEach` inside the `Object.keys.forEach` loop 🤔 

Do you mean inside this `forEach` on line 49 we bring inside the `Object.keys.forEach`? It would mean looping over more iterations altogether (since there's always going to be more attributes than newNodes). Maybe I'm misunderstanding though and need more coffee 😆

Reply:
Could you do the `Object.keys(attributes).forEach((attributeKey)` loop inside the first `newNodes.forEach((node)` loop? Then you could move the styles loop up there too and reduce iterations.

```js
// Make sure all styles are copied to the autocomplete & plain textareas
const cssText = document.defaultView.getComputedStyle(
  originalNodeToReplace,
  '',
).cssText;

const attributes = originalNodeToReplace.attributes;
newNodes.forEach((node) => {
  Object.keys(attributes).forEach((attributeKey) => {
    node.setAttribute(
      attributes[attributeKey].name,
      attributes[attributeKey].value,
    );
  });
  
  node.style.cssText = cssText;
  // Make sure no transition replays when the new textareas are mounted
  node.style.transition = 'none';
  // Copy any initial value
  node.value = originalNodeToReplace.value;
});
```

Reply:
I think that's what @nickytonline was suggesting - thanks @Link2Twenty - my brain hadn't quite woken up this morning 😆

Reply:
> There's only one `forEach` inside the `Object.keys.forEach` loop 🤔 
> 
> 
> 
> Do you mean inside this `forEach` on line 49 we bring inside the `Object.keys.forEach`? It would mean looping over more iterations altogether (since there's always going to be more attributes than newNodes). Maybe I'm misunderstanding though and need more coffee 😆 

Maybe I need more coffee. 😎 I'll double check this. Maybe I was seeing double from the git diff. 😅

Reply:
I've switched it up so that we set the attributes inside the `newNodes.forEach...`. My brain is definitely engaging Friday-mode already 🤣